### PR TITLE
make patching logic more universal

### PR DESCRIPTION
**Pull request by Zackptg5**

### CODE REVIEW

File: config.sh [hunk 1]
```diff
@@ -160,58 +160,28 @@ install_v4a_module() {
...
-    cp -af $NEW_SYS_CONFIG_FILE $CFG
-    sed -i 's/^    <libraries>/    <libraries>\n        <library name="v4a_fx" path="libv4a_fx_ics.so"\/>/g' $CFG
-    sed -i 's/^    <effects>/    <effects>\n        <effect name="v4a_standard_fx" library="v4a_fx" uuid="41d3c987-e6cf-11e3-a88a-11aba5d5c51b"\/>/g' $CFG
-  fi
+	ui_print "* Modifying audio_effects.conf"
+  CFGS="${CFGS} $(find -L /system -type f -name "*audio_effects*.conf")"
+  CFGSXML="${CFGSXML} $(find -L /system -type f -name "*audio_effects*.xml")"
+
+  for FILE in ${CFGS}; do
+    ui_print "* Found $FILE, copying and modifying"
```
Comment (ShadySquirrel) [inline config.sh]: Can you add `if [ -f "$FILE" ];` check? Goal is to not edit symlinks (some ROMs simply symlink /system/etc/audio_effects and /system/vendor/etc/audio_effects).

Same goes for XML block.

Thank you!

File: config.sh [hunk 2]
```diff
@@ -160,58 +160,28 @@ install_v4a_module() {
...
-    ui_print "* Found $NEW_SYS_CONFIG_FILE, copying and modifying"
-    mkdir -p $MODPATH/system/vendor/etc
-    CFG="$MODPATH/$NEW_SYS_CONFIG_FILE"
-    cp -af $NEW_SYS_CONFIG_FILE $CFG
-    sed -i 's/^    <libraries>/    <libraries>\n        <library name="v4a_fx" path="libv4a_fx_ics.so"\/>/g' $CFG
-    sed -i 's/^    <effects>/    <effects>\n        <effect name="v4a_standard_fx" library="v4a_fx" uuid="41d3c987-e6cf-11e3-a88a-11aba5d5c51b"\/>/g' $CFG
-  fi
+	ui_print "* Modifying audio_effects.conf"
+  CFGS="${CFGS} $(find -L /system -type f -name "*audio_effects*.conf")"
+  CFGSXML="${CFGSXML} $(find -L /system -type f -name "*audio_effects*.xml")"
```
Comment (ShadySquirrel) [inline config.sh]: I like this, this actually fixes the 'vendor decided to add prefix/suffix to the standard file' issue I had with some devices (looking at HTC). Great work!

**Comment by Zackptg5** [comment]
The conditional isn't needed since the find command searches for any existing files. Since the find command only searches /system. So how it works is:
For non-vendor partition devices: It'll find the system/vendor audio effects file with the path: /system/vendor/etc/audio_effects.cfg and copy it to $MODPATH/system/vendor/etc/audio_effects.cfg
For vendor partition devices: It'll find the vendor audio effects but since it followed the symlink, the path will be: /system/vendor/etc/audio_effects.cfg and copy it to $MODPATH/system/vendor/etc/audio_effects.cfg which is what we want. The find command doesn't search the actual vendor partition so there won't be any duplications

**Comment by ShadySquirrel** [comment]
I've figured out just as you wrote that comment, had to check flags, last time I've used `find` with anything but `-name` was really long ago.

Thanks for the patches, everything is merged :)

**Pull request merged**